src/i_modmusic.c

```diff
@@ -0,0 +1,149 @@
+//
+// Copyright(C) 2023 Roman Fomin
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//
+
+#ifdef HAVE_MODPLUG
+
+#include <libmodplug/modplug.h>
+
+#include "doomtype.h"
+#include "i_sound.h"
+#include "i_oalmusic.h"
+
+typedef struct
+{
+    ModPlugFile *file;
+
+    void *data;
+    int size;
+} modplug_t;
+
+static modplug_t modplug;
+
+static ModPlug_Settings settings;
+
+static uint32_t MOD_Callback(byte *buffer, uint32_t buffer_samples)
+{
+    int filled;
+
+    filled = ModPlug_Read(modplug.file, buffer, buffer_samples * 4);
+
+    return filled / 4;
```

Comment:
Is this another 8-bit mono / 16-bit stereo calculation?

Reply:
It's 16 bit stereo.